pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -1661,3 +1726,47 @@ func (ctrl *PersistentVolumeController) findDeletablePlugin(volume *v1.Persisten
 	}
 	return plugin, nil
 }
+
+// obtain provisioner/deleter name for a volume
+func (ctrl *PersistentVolumeController) getProvisionerNameFromVolume(volume *v1.PersistentVolume) string {
+	plugin, err := ctrl.findDeletablePlugin(volume)
+	if err != nil {
+		return "N/A"
+	}
+	// if external provisioner was used for provisioning,
+	// the volume MUST have annotation of AnnDynamicallyProvisioned, use the value
+	// as the provisioner name
+	if plugin == nil {
+		return volume.Annotations[pvutil.AnnDynamicallyProvisioned]
```

Comment:
I think in this case we can actually get it from the storage class no problem

Reply:
is it different from using this annotation? my understanding is its equivalent. Getting from storage class of course is another option, however I am trying to avoid the call to v1helper.GetPersistentVolumeClass if not necessary.

Reply:
Using `AnnDynamicallyProvisioned` is still problematic though. As I said commented before - using `AnnDynamicallyProvisioned` here will result in arbitrary provisioner name(which is just node name in case of local-storage-provisioner) and will cause explosion of metrics for anyone using local-storage-provisioner. 

@yuxiangqian you commented:

>"provision-by" is not used in this case for the concerns listed above

Which I do not understand. Local storage plugin (in-tree) does not implement deletable interface and hence `plugin` will be `nil`. So either for local-storage use cases, depending on provisioner name specified in storageclass -either the metrics will be reported with `N/A` name or will cause explosion of metrics with each metric having node name as a dimension.

Reply:
Since local-storage-provisioner tags PVs with `pv.kubernetes.io/provisioned-by=local-volume-provisioner-minikube-18f57fb2-a186-11e7-b543-080027d51893`, this will invariably result in each delete metric being reported separately..

Reply:
Good catch. By saying "provision-by is not used in this case for the concerns listed above" applies to the PVC thread not here. 
I will update the PR for plugin == nil case to use the storage class provisioner instead